scipy/sparse/sputils.py

```diff
@@ -251,6 +251,61 @@ def validateaxis(axis):
             raise ValueError("axis out of range")
 
 
+def check_shape(args, current_shape=None):
+    """Imitate numpy.matrix handling of shape arguments"""
+    int_types = (int, np.int8, np.int16, np.int32, np.int64,
+                 np.uint8, np.uint16, np.uint32, np.uint64,
+                 np.intc, np.intp)
+
+    def check_int(arg):
+        if isinstance(arg, int_types):
+            return int(arg)
+        elif isinstance(arg, np.ndarray) and np.isscalar(arg) and \
+                arg.dtype in int_types:
+            return int(arg)
+        else:
+            raise TypeError("'{}' object cannot be interpreted as an "
+                            "integer".format(args[0].__class__.__name__))
+
+    if len(args) == 0:
+        raise TypeError("'reshape' missing 1 required positional argument: "
```

Comment:
This function isn't always called from `reshape`, so this error message could be misleading.

Reply:
Technically, only the `reshape` function can have zero args, but we might want to reuse this function for other shape checks, so I'll make it more generic.